File: gunhoflash/valid-parentheses.py
class Solution:
  def isValid(self, s: str) -> bool:
    stack = []
    for char in s:
      match char:
        case '(' | '{' | '[':
          stack.append(char)
        case ')':
          if len(stack) == 0 or stack.pop() != '(':
            return False
        case '}':
          if len(stack) == 0 or stack.pop() != '{':
            return False
        case ']':
          if len(stack) == 0 or stack.pop() != '[':
            return False
    return len(stack) == 0

File: gunhoflash/valid-parentheses.ts
function isValid(s: string): boolean {
  const stack: string[] = [];
  for (const char of s) {
    switch (char) {
      case '(':
      case '{':
      case '[':
        stack.push(char);
        break;
      case ')':
        if (stack.pop() !== '(') {
          return false;
        }
        break;
      case '}':
        if (stack.pop() !== '{') {
          return false;
        }
        break;
      case ']':
        if (stack.pop() !== '[') {
          return false;
        }
        break;
    }
  }
  return stack.length === 0;
}
